packages/core/src/shared/resourcefetcher/node/httpResourceFetcher.ts

```diff
@@ -0,0 +1,129 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs' // eslint-disable-line no-restricted-imports
+import * as http from 'http'
+import * as https from 'https'
+import * as stream from 'stream'
+import got, { RequestError } from 'got'
```

Comment:
Why do we need a node-only version? We are trying to eliminate "got" entirely.

Reply:
IIRC @nkomonen-amazon mentioned that fs streams from fetch weren't compatible with fs streams from node yet. Right now this will limit `got` to this single file until things are compatible and then I can remove all the other references of got

Reply:
From an old Web quip:

* Node.js has their own [Stream implementation](https://nodejs.org/docs/latest-v18.x/api/stream.html#stream)
* Browser/Web uses a different [Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API) (implements the [WHATWG Streams Standard](https://streams.spec.whatwg.org/))
  * [Though this is supported in Node 21+](https://nodejs.org/api/webstreams.html#web-streams-api) and 'got' could technically update to this API